gammapy/spectrum/results.py

```diff
@@ -0,0 +1,467 @@
+
+        # Get fixed parameters from model (not stored in covar)
+        for par in model.pars:
+            if par.name in thawed_pars:
+                continue
+            if par.name == 'ref':
+                name = 'reference'
+                unit = Unit('keV')
+            parameters[name] = par.val * unit
+            parameter_errors[name] = 0 * unit
+
+        fluxes = Bunch()
+        fluxes['1TeV'] = model(1e6) * Unit('cm-2 s-1')
+        flux_errors = Bunch()
+        flux_errors['1TeV'] = 0 * Unit('cm-2 s-1')
+
+        flux_points = Table(data=flux_graph, masked=True)
+        flux_points['energy'].unit = 'keV'
+        # Fill empty error columns to be consistent
+        val = np.zeros(shape=flux_points['energy'].shape)
+        flux_points['energy_err_hi'] = Column(data=val, unit='keV')
+        flux_points['energy_err_lo'] = Column(data=val, unit='keV')
+        flux_points['flux'].unit = 'cm-2 s-1 keV-1'
+        flux_points['flux_err_hi'].unit = 'cm-2 s-1 keV-1'
+        flux_points['flux_err_lo'].unit = 'cm-2 s-1 keV-1'
+
+        return cls(energy_range=energy_range, parameters=parameters,
+                   parameter_errors=parameter_errors,
+                   spectral_model=spectral_model, flux_points=flux_points,
+                   fluxes=fluxes, flux_errors=flux_errors)
+
+    def to_yaml(self):
+        """Export spectrum results to YAML dict"""
+        import yaml
+        val = dict()
+        val['energy_range'] = dict(min=self.energy_range[0].value,
+                                   max=self.energy_range[1].value,
+                                   unit='{}'.format(self.energy_range.unit))
+        val['parameters'] = dict()
+        for key in self.parameters:
+            par = self.parameters[key]
+            err = self.parameter_errors[key]
+            val['parameters'][key] = dict(value=par.value,
+                                          error=err.value,
+                                          unit='{}'.format(par.unit))
+        val['spectral_model'] = self.spectral_model
+        val['flux_points'] = dict()
+
+        if self.flux_points is not None:
+            fp = self.flux_points
+            for col in fp.columns:
+                val['flux_points'][col] = [float(_) for _ in fp[col]]
+            val['flux_points']['energy_unit'] = '{}'.format(fp['energy'].unit)
+            val['flux_points']['flux_unit'] = '{}'.format(fp['flux'].unit)
+
+        rtval = yaml.safe_dump(val, default_flow_style=False)
+        return rtval
+
+    def write(self, filename):
+        """Write YAML file
+
+        TODO: Round float to a given precision
+
+        Parameters
+        ----------
+        filename : str
+            File to write
+        """
+        val = self.to_yaml()
+        with open(str(filename), 'w') as outfile:
+            outfile.write(val)
+
+    @classmethod
+    def from_yaml(cls, val):
+        """Create `~gammapy.spectrum.results.SpectrumResult` from YAML dict
```

Comment:
This method doesn't do YAML de-serialisation, it just takes a dict, no?
If so, call the method `from_dict` and make the docstring correct` (it currently says "from YAML dict").